tests/device_selector/device_selector_aspect.cpp

```diff
@@ -0,0 +1,469 @@
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+*******************************************************************************/
+
+#include "../common/common.h"
+#include "../common/disabled_for_test_case.h"
+#include "../common/random.h"
+#include "../common/type_coverage.h"
+
+#include <random>
+#include <sstream>
+
+namespace device_selector_aspect {
+using namespace sycl_cts;
+
+/** List of all device aspects. */
+#define ASPECT_LIST                                                            \
+  sycl::aspect::cpu, sycl::aspect::gpu, sycl::aspect::accelerator,             \
+      sycl::aspect::custom, sycl::aspect::host_debuggable, sycl::aspect::fp16, \
+      sycl::aspect::fp64, sycl::aspect::atomic64, sycl::aspect::image,         \
+      sycl::aspect::online_compiler, sycl::aspect::online_linker,              \
+      sycl::aspect::queue_profiling, sycl::aspect::usm_device_allocations,     \
+      sycl::aspect::usm_host_allocations,                                      \
+      sycl::aspect::usm_atomic_host_allocations,                               \
+      sycl::aspect::usm_shared_allocations,                                    \
+      sycl::aspect::usm_atomic_shared_allocations,                             \
+      sycl::aspect::usm_system_allocations
+
+/** Return a list of all defined aspects. */
+static std::vector<sycl::aspect> get_aspect_list() {
+#ifdef SYCL_CTS_COMPILING_WITH_DPCPP
+  WARN(
+      "DPCPP does not define sycl::aspect::emulated."
+      "Skipping test cases for this aspect.");
+#endif
+  static const std::vector<sycl::aspect> types = {ASPECT_LIST
+#ifndef SYCL_CTS_COMPILING_WITH_DPCPP
+                                                  ,
+                                                  sycl::aspect::emulated
+#endif
+  };
+  return types;
+}
+
+/** Return a named value pack of all defined aspects. */
+static auto get_aspect_pack() {
+#ifdef SYCL_CTS_COMPILING_WITH_DPCPP
+  WARN(
+      "DPCPP does not define sycl::aspect::emulated."
+      "Skipping test cases for this aspect.");
+#endif
+  static const auto types = value_pack<sycl::aspect, ASPECT_LIST
+#ifndef SYCL_CTS_COMPILING_WITH_DPCPP
+                                       ,
+                                       sycl::aspect::emulated
+#endif
+                                       >::generate_named();
+  return types;
+}
+
+/**
+ * Check whether all specified constructors for \p aspect_selector are
```

Comment:
I'm not sure what you mean, are you suggesting to use
```c++
/**
 comment
 */
```
instead of 
```c++
/**
 * comment
 */
```
The latter style is prevalent in the CTS.

Reply:
Yes :-) and yes :-(
This is just visual noise.
Probably we need to update/have a coding style.
Not blocking.

Reply:
I have applied it.

Reply:
I have applied it in my other open PRs as well.